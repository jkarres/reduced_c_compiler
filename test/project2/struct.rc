
structdef S {
  int i;
  float f;
  bool b;
  float[5] ia;
};

S[10] sa;


function : void main() { 

  S s;
  s.f = s.i = 42;
  s.b = true;
  s.ia[3] = 9.7;

  cout << s.f << endl;
  cout << s.i << endl;
  cout << s.b << endl;
  cout << s.ia[3] << endl;


  int i, j;

  for (i = 0; i < 10; ++i) {
    for (j = 0; j < 5; ++j) {
      sa[i].ia[j] = j / (i+1.00);
    }
  }


  for (i = 10 - 1; i >= 0; --i) {
    for (j = 5-1; j >= 0; --j) {
      cout << sa[i].ia[j] << endl;
    }
  }

}
