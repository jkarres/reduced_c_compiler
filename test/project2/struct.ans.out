42.00
42
true
9.70
0.40
0.30
0.20
0.10
0.00
0.44
0.33
0.22
0.11
0.00
0.50
0.38
0.25
0.12
0.00
0.57
0.43
0.29
0.14
0.00
0.67
0.50
0.33
0.17
0.00
0.80
0.60
0.40
0.20
0.00
1.00
0.75
0.50
0.25
0.00
1.33
1.00
0.67
0.33
0.00
2.00
1.50
1.00
0.50
0.00
4.00
3.00
2.00
1.00
0.00
